### DISCUSSION TITLE
blog/traffic-interception-with-geneve-tunnel-with-istio-ambient-mesh/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/rootsongjc.github.io#331] [Announcements]
# blog/traffic-interception-with-geneve-tunnel-with-istio-ambient-mesh/

本文介绍了什么是 Geneve 隧道，与 VXLAN 的区别，以及它在 Istio Ambient Mesh 中的应用，最后谈及使用 eBPF 优化流量拦截。

https://jimmysong.io/blog/traffic-interception-with-geneve-tunnel-with-istio-ambient-mesh/

**Comment by BSWANG** [comment]
大佬，有些问题请教下：
1. 为什么需要隧道来劫持到ambient，既然ztunnel本身的网卡就是veth的，应该可以直接在策略路由表中将默认网关指向为ztunnel的IP来将流量路由到ztunnel中吧
2. 目前的iptables应该还会设置connmark来跟踪，看起来只有首包会走到istioout和istioin， 后续的包会走到0x40的mark直接走vether不封包了。抓包的话也可以看到只有SYN到了istioin的网卡，想了解下这块的设计的原因是什么。

**Reply by BSWANG** [reply]
另外还有ambient是否使用host网络和按照ipset做tproxy会更简单些？

**Reply by rootsongjc** [reply]
现在这种劫持模式也落伍了，Istio 社区推荐使用 [Istio CNI](https://github.com/istio/istio/tree/master/cni) 来实现了。
